File: 2295-replace-elements-in-an-array/2295-replace-elements-in-an-array.rs
use std::collections::HashMap;

impl Solution {
    pub fn array_change(mut nums: Vec<i32>, operations: Vec<Vec<i32>>) -> Vec<i32> {
        let mut map = HashMap::new();
        
        for i in 0..nums.len(){
            map.insert(nums[i], i);
        }
        
        for v in operations{
            let old = v[0];
            let new = v[1];
            
            let i = map.remove(&old).unwrap();
            map.insert(new, i);
            nums[i] = new;
        }
        
        nums
    }
}

File: 2295-replace-elements-in-an-array/2295-replace-elements-in-an-array.py
class Solution:
    def arrayChange(self, nums: List[int], operations: List[List[int]]) -> List[int]:
        map = {}
        
        for (i, n) in enumerate(nums):
            map[n] = i
            
        for [old, new] in (operations):
            i = map.pop(old)
            map[new] = i
            
        res = [0 for _ in range(0, len(map))]
        
        for (n, i) in map.items():
            res[i] = n
        
        return res
        